pkg/chunkenc/unordered.go

```diff
@@ -346,14 +369,29 @@ func (hb *unorderedHeadBlock) Serialise(pool WriterPool) ([]byte, error) {
 		logproto.FORWARD,
 		0,
 		math.MaxInt64,
-		func(ts int64, line string) error {
+		func(ts int64, line string, metaLabels labels.Labels) error {
 			n := binary.PutVarint(encBuf, ts)
 			inBuf.Write(encBuf[:n])
 
 			n = binary.PutUvarint(encBuf, uint64(len(line)))
 			inBuf.Write(encBuf[:n])
 
 			inBuf.WriteString(line)
+
+			if hb.format >= UnorderedWithMetadataHeadBlockFmt {
+				// Serialize metadata labels
+				n = binary.PutUvarint(encBuf, uint64(len(metaLabels)))
```

Comment:
Good point. We can do it here, but we'd need to iterate through all the labels and sum up their lengths. With strings interning, IIUC, the label and values would be refs to wherever the actual string is. Therefore, in that case we wouldn't need to iterate through all the label value pairs, but rather multiply `len(metaLabels)` by twice the ref size, right?

In that case, I think it would make more sense to do it in the PR for strings interning directly? To not complicate this PR more than necessary provided that this part would change. Wdyt?

Reply:
Keep in mind, this will break compatibility between this PR and a future PR unless we add a newer format than `UnorderedWithMetadataHeadBlockFmt`. This is one of the benefits of not setting this as the default yet (via `DefaultHeadBlockFmt = UnorderedHeadBlockFmt`) -- we can change the implementation in a later PR before we start using it.